Command line interface: reworked args including adding json file arg

davidlunarg opened the pull request.

davidlunarg committed:
Command line interface: reworked args including adding json file arg

panos-lunarg commented on tools/replay/CMakeLists.txt:
```diff
@@ -37,14 +37,30 @@ target_sources(gfxrecon-replay
 
-target_include_directories(gfxrecon-replay PUBLIC ${CMAKE_BINARY_DIR})
+if (WIN32)
+    target_include_directories(gfxrecon-replay PUBLIC ${CMAKE_BINARY_DIR} ../../external/json/win64/include)
```
I think `${CMAKE_BINARY_DIR} ../../external/json/win64/include` is the wrong directory.
It should be `${CMAKE_SOURCE_DIR}/external/json/linux/include`
Same for the else case

panos-lunarg commented on tools/replay/CMakeLists.txt:
```diff
@@ -37,14 +37,30 @@ target_sources(gfxrecon-replay
+if (WIN32)
+    target_link_directories(gfxrecon_decode
+                            PUBLIC
+                                ${CMAKE_SOURCE_DIR}/external/json/win64/lib)
```
We usually include pre-compiled libraries and the header files in `external/precompiled/` directory.
I think headers + binaries should be moved there for consistency

panos-lunarg commented on tools/replay/CMakeLists.txt:
```diff
@@ -37,14 +37,30 @@ target_sources(gfxrecon-replay
                           gfxrecon_graphics
                           gfxrecon_format
                           gfxrecon_util
+                          jsoncpp.lib
```
This should be just `jsoncpp` in order to work for Linux/Android too

panos-lunarg committed:
Moved jsoncpp lib to appropriate folders

Moved jsoncpp under the appropriate external/precompiled directories
Added pre-compiled binaries for linux 32bit, android 32 and android 64

davidlunarg committed:
Update spirv-reflect header file

panos-lunarg commented on USAGE_desktop_Vulkan.md:
@davidlunarg 
This looks good.
I think it would be better if we devote a new sub-chapter under chapter `2. Replaying API Calls` called something like `2.2 Dumping Resources` where we include a more detailed information and more importantly examples of command line arguments and an input json example

panos-lunarg commented on framework/decode/vulkan_replay_options.h:
```diff
@@ -463,17 +454,19 @@ struct VulkanReplayOptions : public ReplayOptions
+    std::vector<uint64_t>                           BeginCommandBuffer_Indices;
+    std::vector<std::vector<uint64_t>>              CmdDraw_Indices;
+    std::vector<std::vector<std::vector<uint64_t>>> RenderPass_Indices;
+    std::vector<std::vector<std::vector<uint64_t>>> NextSubPass_Indices;
```
Please see comment in relevant internal document regarding render pass related indices
